### PR TITLE
Adding some methods for FbxAnimCurve, FbxAnimCurveBase, FbxCollection…

### PR DESCRIPTION
**Pull request by utjkaz**
I'd like to make FBX editor on Unity. However it's not enough the implementations for me.

FbxCollection and FbxGeometryBase
Use for mesh editor.

FbxAnimCurve and FbxAnimCurveBase
Use for animation curve editor.

### CODE REVIEW

**Comment by vkovec** [comment]
Awesome! would it be possible to please add some simple unit tests for these functions?

For FbxAnimCurve, you can add them here: https://github.com/Unity-Technologies/com.autodesk.fbx/blob/master/tests/RuntimeTests/Editor/UnitTests/FbxAnimCurveTest.cs#L62
For FbxCollection: https://github.com/Unity-Technologies/com.autodesk.fbx/blob/master/tests/RuntimeTests/Editor/UnitTests/FbxCollectionTest.cs#L16
For FbxGeometryBase: https://github.com/Unity-Technologies/com.autodesk.fbx/blob/master/tests/RuntimeTests/Editor/UnitTests/FbxGeometryBaseTest.cs#L16

In order to run the tests locally, after building the project and adding the package in `build/install` to a Unity project, mark the `com.autodesk.fbx` package as testable in the Unity project's manifest.json file: https://docs.unity3d.com/Manual/cus-tests.html#tests

Then you should be able to run the tests from Window > General > Test Runner. Thank you!

**Comment by utjkaz** [comment]
Okay! Thank you for supporting me, I'll make unit tests & commit again later.

**Comment by utjkaz** [comment]
Okay, I've added unit tests for all added functions.

File: tests/RuntimeTests/Editor/UnitTests/FbxAnimCurveTest.cs [hunk 1]
```diff
@@ -57,6 +57,18 @@ protected override void TestSceneContainer()
             }
         }
 
+        public static bool AssertSimilar(double expected, double actual,
+                double tolerance = 1e-10, bool nothrow = false)
+        {
```
Comment (vkovec) [inline tests/RuntimeTests/Editor/UnitTests/FbxAnimCurveTest.cs:62]: I didn't see the nothrow parameter or return value used when calling the function. Would assert with a tolerance passed in work? `Assert.AreEqual(double expected, double actual, double tolerance);`

File: tests/RuntimeTests/Editor/UnitTests/FbxAnimCurveTest.cs [hunk 2]
```diff
@@ -83,7 +95,64 @@ public void TestBasics ()
...
+                int startIndex = curve.KeyAdd (FbxTime.FromSecondDouble (4));
+                int endIndex = curve.KeyAdd (FbxTime.FromSecondDouble (5));
+                curve.KeyRemove (startIndex, endIndex);
+                Assert.AreEqual (1, curve.KeyGetCount ());
+
+                // test KeySetTime
+                curve.KeySetTime (index, FbxTime.FromSecondDouble (4));
+                Assert.AreEqual (FbxTime.FromSecondDouble(4), curve.KeyGetTime (index));
+
+				// test KeyClear
```
Comment (vkovec) [inline tests/RuntimeTests/Editor/UnitTests/FbxAnimCurveTest.cs:111]: please fix up spacing here

Comment (utjkaz) [reply tests/RuntimeTests/Editor/UnitTests/FbxAnimCurveTest.cs]: Okay, I've fixed it.

File: tests/RuntimeTests/Editor/UnitTests/FbxAnimCurveTest.cs (hunk 1 above)
Comment (utjkaz) [reply tests/RuntimeTests/Editor/UnitTests/FbxAnimCurveTest.cs]: Indeed, it's better to use default Assert.AreEqual with tolerance. I've fixed these tests with it.

**Review by benoithudson — APPROVED** [review]
Looks great, thanks!

**Review by vkovec — APPROVED** [review]

**Pull request merged**